src/lwpa_plat/windows/lwpa/socket.c

```diff
@@ -669,45 +881,82 @@ int lwpa_poll(LwpaPollfd *fds, size_t nfds, int timeout_ms)
+      lwpa_mutex_give(&context->lock);
+    }
+    return res;
+  }
+}
+
+lwpa_error_t handle_select_result(LwpaPollContext *context, LwpaPollEvent *event, const LwpaPollFdSet *readfds,
+                                  const LwpaPollFdSet *writefds, const LwpaPollFdSet *exceptfds)
+{
+  // Init the event data.
+  event->socket = LWPA_SOCKET_INVALID;
+  event->events = 0;
+  event->err = kLwpaErrOk;
+
+  // The default return; if we don't find any sockets set that we passed to select(), something has
+  // gone wrong.
+  lwpa_error_t res = kLwpaErrSys;
 
-      if (FD_ISSET(fd->fd, &readfds) || FD_ISSET(fd->fd, &writefds) || FD_ISSET(fd->fd, &exceptfds))
+  for (LwpaPollCtxSocket *sock_desc = context->sockets; sock_desc < context->sockets + context->socket_arr_size;
+       ++sock_desc)
+  {
+    if (sock_desc->socket == LWPA_SOCKET_INVALID)
+      continue;
+
+    if (LWPA_FD_ISSET(sock_desc->socket, readfds) || LWPA_FD_ISSET(sock_desc->socket, writefds) ||
+        LWPA_FD_ISSET(sock_desc->socket, exceptfds))
+    {
+      res = kLwpaErrOk;
+      event->socket = sock_desc->socket;
+      event->user_data = sock_desc->user_data;
+
+      /* Check for errors */
+      int error;
+      int error_size = sizeof(error);
+      if (getsockopt(sock_desc->socket, SOL_SOCKET, SO_ERROR, (char *)&error, &error_size) == 0)
       {
-        /* Check for errors */
-        int error;
-        int error_size = sizeof(error);
-        if (getsockopt(fd->fd, SOL_SOCKET, SO_ERROR, (char *)&error, &error_size) == 0)
+        if (error != 0)
         {
-          if (error != 0)
-          {
-            fd->revents |= LWPA_POLLERR;
-            fd->err = err_plat_to_lwpa(error);
-          }
-        }
-        else
-        {
-          return (int)err_plat_to_lwpa(WSAGetLastError());
+          event->events |= LWPA_POLL_ERR;
+          event->err = err_plat_to_lwpa(error);
         }
       }
-      if (nreadfds && (fd->events & LWPA_POLLIN) && FD_ISSET(fd->fd, &readfds))
+      else
+      {
+        res = err_plat_to_lwpa(WSAGetLastError());
+        break;
+      }
+      if (LWPA_FD_ISSET(sock_desc->socket, readfds))
       {
-        fd->revents |= LWPA_POLLIN;
+        if (sock_desc->events & LWPA_POLL_IN)
+          event->events |= LWPA_POLL_IN;
       }
-      if (nwritefds && (fd->events & LWPA_POLLOUT) && FD_ISSET(fd->fd, &writefds))
+      if (LWPA_FD_ISSET(sock_desc->socket, writefds))
       {
-        fd->revents |= LWPA_POLLOUT;
+        if (sock_desc->events & LWPA_POLL_CONNECT)
+          event->events |= LWPA_POLL_CONNECT;
+        else if (sock_desc->events & LWPA_POLL_OUT)
+          event->events |= LWPA_POLL_OUT;
       }
-      if (nexceptfds && (fd->events & LWPA_POLLPRI) && FD_ISSET(fd->fd, &exceptfds))
+      if (LWPA_FD_ISSET(sock_desc->socket, exceptfds))
       {
-        fd->revents |= LWPA_POLLPRI;
+        if (sock_desc->events & LWPA_POLL_CONNECT)
```

Comment:
I'm a bit confused by LWPA_POLL_CONNECT.  When does that event get cleared?  Otherwise the user will never see LWPA_POLL_OOB, right?